### ISSUE TITLE
Feature - Adding Dataframe applicability to some Series string methods

### ISSUE THREAD

**Issue by kbhutani0001** [pandas-dev/pandas#22911]
Hello,
Initially the string methods, like replace, lower, zfill, strip etc etc.. are restricted to Series use only.
It would be good if a parameter is put to use it on data frames too. Methods like strip won't affect numeric columns since they wount be having spaces already. But if there is a method which can affect a numeric column, it can be excluded using exclude parameter (which should be added). 
A simple way of doing it is demonstrated below. 

```
import pandas as pd
data=pd.read_csv("nba.csv")
data
dtypes=data.dtypes.astype(str)
for columns in data.columns:
    data[columns]=data[columns].astype(str)
    data[columns]=data[columns].str.replace(" ","")
    data[columns]=data[columns].astype(dtypes[columns])
data
```
If this issue is approved, I would like to work and contribute to this feature.

**Comment by TomAugspurger** [comment]
It's not clear what should happen when you have a mix of numeric and string dtypes.

We typically recommend something like

```python
columns = data.select_dtypes("str")
data[column] = data[columns].apply(pd.Series.str.replace(" ", ","))
```

for applying the same column-wise transformation to a subset of the columns.

**Comment by kbhutani0001** [comment]
It's not clear that what anyone would do with zfill on a series of strings
too. Practically it's more of a numeric series method but still applicable
on just string series. My point is to make those methods available for
either numeric series too or whole data frame which would be better.

**Comment by TomAugspurger** [comment]
Ah, in that case I think that https://github.com/pandas-dev/pandas/issues/17211 may cover everything you're asking for here. Can you confirm?

**Comment by kbhutani0001** [comment]
Not really, I am proposing a parameter to string methods so that they can
be applied to data frames directly too. It has nothing to do with .astype()
.astype() is a part of it since the whole data would be converted to str
before applying string operation and back to orignal data type again.

**Comment by kbhutani0001** [comment]
Many methods, for example zfill()
Doesn't have a practical use on a string series. Why'd someone put some
zeroes to a string. In 90% of cases, it is a numeric series which they
convert to string and then apply the method and convert to int back again.
Similarly many methods like this are restricted to string use only and
doesn't have much of a practical use.

Knowing that, if there is an option to apply the whole method on a data
frame itself
Instead of doing it separately for each string can help a lot

**Comment by TomAugspurger** [comment]
That discussion moved on from `.astype` to a more dedicated `.format` method
